bibliopixel/main/template/animation.py.tmpl

```diff
@@ -0,0 +1,45 @@
+from bibliopixel import animation
+from bibliopixel.colors import COLORS
+
+"""
+This is an example Animation class which has a single field -
+`color` with the default value `COLORS.green`.
+
+You can edit this as you like, add and remove fields anda
+do Pythonic things
+"""
+
+class {class_name}(animation.BaseAnimation):
+    def __init__(
+            self, *args,
+            # The fields for your class go here!
+            color=COLORS.green,
+
+            # End of the fields for your class
+            **kwds):
+        super().__init__(*args, **kwds)
+        self.color = color
+        # Your initialization code goes here.
+
+    def step(self, amt=1):
+        this_pixel = self.cur_step % len(self.color_list)
```

Comment:
but wait, i'm already confused as a newbie. We jut set a variable called `self.color` and now we are finding the length of `self.color_list` Clearly one of these variables (at least) comes from a parent class. Do we always have access to self.color_list? If we set `self.color` does it automatically go to the `color_list` etc.